frame/staking/src/lib.rs

```diff
@@ -2580,13 +2452,16 @@ impl<T: Trait> Module<T> {
 		election_size: ElectionSize,
 	) -> DispatchResultWithPostInfo {
 		// Do the basic checks. era, claimed score and window open.
-		Self::pre_dispatch_checks(claimed_score, era)?;
-		// the weight that we will refund in case of a correct submission. We compute this now
-		// because the data needed for it will be consumed further down.
-		let adjusted_weight = weight::weight_for_correct_submit_solution::<T>(
-			&winners,
-			&compact_assignments,
-			&election_size,
+		let _ = Self::pre_dispatch_checks(claimed_score, era)?;
+
+		// before we read any further state, we check that the unique targets in compact is same as
```

Comment:
@thiolliere this check is basically mitigating your grumble in the previous rounds of weights: that an invalid solution could have so many potential targets, for each of which we read `SlashingSpans`. But here we check that we have only ever exactly `winners.len()` targets in `compact`.